internal/topology/variables/clusterclass_variable_validation.go

```diff
@@ -178,20 +152,43 @@ func validateSchema(schema *clusterv1.JSONSchemaProps, fldPath *field.Path) fiel
 		return append(allErrs, validationErrors...)
 	}
 
-	// Validate defaults in the structural schema.
-	validationErrors, err := structuraldefaulting.ValidateDefaults(fldPath.Child("schema"), ss, true, true)
```

Comment:
This is a recursive function itself so each default value was being recursed twice producing one defaulting error (if there was one) for each level of nesting in the object)

Reply:
I think we should keep it but it shouldn't be part of our recursion